src/Sylius/Behat/Context/Ui/Admin/ManagingChannelsContext.php

```diff
@@ -0,0 +1,409 @@
+
+        Assert::true(
+            $currentPage->checkValidationMessageFor($element, sprintf('Please enter channel %s.', $element)),
+            sprintf('Tax category %s should be required.', $element)
+        );
+    }
+
+    /**
+     * @Given I want to modify a channel :channel
+     * @Given /^I want to modify (this channel)$/
+     */
+    public function iWantToModifyChannel(ChannelInterface $channel)
+    {
+        $this->updatePage->open(['id' => $channel->getId()]);
+    }
+
+    /**
+     * @Then /^(this channel) name should be "([^"]+)"$/
+     * @Then /^(this channel) should still be named "([^"]+)"$/
+     */
+    public function thisChannelNameShouldBe(ChannelInterface $channel, $channelName)
+    {
+        $this->iWantToBrowseChannels();
+
+        Assert::true(
+            $this->indexPage->isResourceOnPage(
+                [
+                    'code' => $channel->getCode(),
+                    'name' => $channelName,
+                ]
+            ),
+            sprintf('Channel name %s has not been assigned properly.', $channelName)
+        );
+    }
+
+    /**
+     * @When I save my changes
+     * @When I try to save my changes
+     */
+    public function iSaveMyChanges()
+    {
+        $this->updatePage->saveChanges();
+    }
+
+    /**
+     * @Then I should be notified that channel with this code already exists
+     */
+    public function iShouldBeNotifiedThatChannelWithThisCodeAlreadyExists()
+    {
+        Assert::true(
+            $this->createPage->checkValidationMessageFor('code', 'Channel code has to be unique.'),
+            'Unique code violation message should appear on page, but it does not.'
+        );
+    }
+
+    /**
+     * @Then there should still be only one channel with :element :value
+     */
+    public function thereShouldStillBeOnlyOneChannelWithCode($element, $value)
+    {
+        $this->iWantToBrowseChannels();
+
+        Assert::true(
+            $this->indexPage->isResourceOnPage([$element => $value]),
+            sprintf('Channel with %s %s cannot be founded.', $element, $value)
+        );
+    }
+
+    /**
+     * @When /^I want to browse channels$/
+     */
+    public function iWantToBrowseChannels()
+    {
+        $this->indexPage->open();
+    }
+
+    /**
+     * @Then I should see :numberOfChannels channels in the list
+     */
+    public function iShouldSeeChannelsInTheList($numberOfChannels)
+    {
+        $foundRows = $this->indexPage->countItems();
+
+        Assert::true(
+            ((int) $numberOfChannels) === $foundRows,
+            sprintf('%s rows with channels should appear on page, %s rows has been found', $numberOfChannels, $foundRows)
+        );
+    }
+
+    /**
+     * @Then the code field should be disabled
+     */
+    public function theCodeFieldShouldBeDisabled()
+    {
+        Assert::true(
+            $this->updatePage->isCodeDisabled(),
```

Comment:
Shouldn't we have logic about checking the state of element in context and pages should only provide states without assert it?